### PR TITLE
Add GitHub action and Makefile target for Windows installer

### PR DESCRIPTION
**Pull request by stweil**
The new GitHub action "installer-for-windows" builds an NSIS installer with Tesseract for Windows.
It is not run automatically, but must be triggered manually.

Autoconf builds also have a new target `winsetup`, so the installer can be built with `make winsetup`.
This requires a build environment with all needed components (NSIS, Windows libraries, ...).

### CODE REVIEW

**Comment by stweil** [comment]
Example: https://github.com/stweil/tesseract/actions/runs/11620260190.

File: nsis/winpath.exe
Comment (egorpugin) [inline nsis/winpath.exe:1]: an .exe?

Comment (stweil) [reply nsis/winpath.exe:1]: You are right, I'll add a rule to build it from winpath.cpp.

Comment (stweil) [reply nsis/winpath.exe:1]: Egor, could you please have a look on the sw build failures?

File: nsis/plugins/INetC.dll
Comment (egorpugin) [inline nsis/plugins/INetC.dll:1]: Is this .dll also needed?

File: nsis/winpath.exe
Comment (egorpugin) [reply nsis/winpath.exe:1]: Yes, something with new python 3.13.

File: nsis/plugins/INetC.dll
Comment (stweil) [reply nsis/plugins/INetC.dll:1]: It's an extension for the NSIS installer and used for the download of Tesseract models.

File: nsis/winpath.exe
Comment (stweil) [reply nsis/winpath.exe:1]: I updated the rules to build winpath.exe, so this binary is no longer in the git repository.

File: nsis/plugins/INetC.dll
Comment (egorpugin) [reply nsis/plugins/INetC.dll:1]: Is it possible to download via curl? Curl is present in modern windows.

Comment (stweil) [reply nsis/plugins/INetC.dll:1]: I updated Makefile.am to download the zip file and unzip the INetC.dll.

**Review by egorpugin — APPROVED** [review]

**Comment by stweil** [comment]
Here is the first result: https://github.com/tesseract-ocr/tesseract/actions/runs/11641358702.

**Pull request merged**